CMakeLists.txt

```diff
@@ -987,6 +996,14 @@ foreach(hal ${OpenCV_HAL})
     else()
       message(STATUS "Carotene: NEON is not available, disabling carotene...")
     endif()
+  elseif(hal STREQUAL "halrvv")
+    if(CV_GCC)
```

Comment:
I believe we don't need GCC check here anymore since we are considering this to be universal RVV HAL.